core/silkworm/types/transaction.cpp

```diff
@@ -375,21 +375,22 @@ namespace rlp {
 }  // namespace rlp
 
 void Transaction::recover_sender() {
-    from.reset();
-
+    if (from.has_value()) {
```

Comment:
IMHO this memoization is a bit dangerous. What if the transaction is modified between invocations of `recover_sender`.

Reply:
Should this be the case we'd be doing something very wrong elsewhere. Removing the check would invoke an expensive extra check of the signature